plugins/module_utils/network/sonic/config/static_routes/static_routes.py

```diff
@@ -0,0 +1,341 @@
+            if interface and next_hop and network_instance:
+                idx = interface + '_' + next_hop + '_' + network_instance
+            if next_hop and not interface and not network_instance:
+                idx = next_hop
+            if next_hop and network_instance and not interface:
+                idx = next_hop + '_' + network_instance
+
+        return idx
+
+    def get_delete_static_routes_requests(self, commands, have, is_delete_all):
+        requests = []
+        if is_delete_all:
+            requests = self.get_delete_all_static_routes_requests(commands)
+        else:
+            for cmd in commands:
+                vrf_name = cmd.get('vrf_name', None)
+                static_list = cmd.get('static_list', [])
+                for static in static_list:
+                    prefix = static.get('prefix', None)
+                    next_hops = static.get('next_hops', [])
+
+                    for cfg in have:
+                        cfg_vrf_name = cfg.get('vrf_name', None)
+                        if vrf_name == cfg_vrf_name:
+                            cfg_static_list = cfg.get('static_list', [])
+                            for cfg_static in cfg_static_list:
+                                cfg_prefix = cfg_static.get('prefix', None)
+                                if prefix == cfg_prefix:
+                                    if prefix and not next_hops:
+                                        requests.append(self.get_delete_static_routes_prefix_request(vrf_name, prefix))
+                                    else:
+                                        for next_hop in next_hops:
+                                            index = next_hop.get('index', {})
+                                            idx = self.generate_index(index)
+                                            metric = next_hop.get('metric', None)
+                                            track = next_hop.get('track', None)
+                                            tag = next_hop.get('tag', None)
+
+                                            cfg_next_hops = cfg_static.get('next_hops', [])
+                                            if cfg_next_hops:
+                                                for cfg_next_hop in cfg_next_hops:
+                                                    cfg_index = cfg_next_hop.get('index', {})
+                                                    cfg_idx = self.generate_index(cfg_index)
+                                                    if idx == cfg_idx:
+                                                        cfg_metric = cfg_next_hop.get('metric', None)
+                                                        cfg_track = cfg_next_hop.get('track', None)
+                                                        cfg_tag = cfg_next_hop.get('tag', None)
+                                                        if idx and not metric and not track and not tag:
```

Comment:
How can "idx" not exist here?

Reply:
there's no not in front of idx

Reply:
What I meant was that "if idx and not metric ..." could just be "if not metric ..." because idx is known to exist.